lib/widgets/witnet/transactions/value_transfer/create_dialog_box/vtt_builder/01_recipient_step.dart

```diff
@@ -72,140 +79,104 @@ class RecipientStepState extends State<RecipientStep>
-            });
-          }
-        } else {
-          setState(() {
-            _errorAddressText = 'Invalid address';
-          });
-        }
-      }
-    }
-    return _errorAddressText != null ? false : true;
+  String? _validateAddress(String? input) {
+    String? errorText;
+    errorText = validateAddress(_address);
+    return errorText;
   }
 
-  bool _validateAmount({force = false}) {
-    if (this.mounted) {
-      if (force || _notFocusForm()) {
-        setState(() {
-          _errorAmountText = null;
-        });
-        if (_amount.isEmpty) {
-          setState(() {
-            _errorAmountText = 'This field is required';
-          });
-        } else if (!_isValidNumber(_amount)) {
-          setState(() {
-            _errorAmountText = 'Invalid number';
-          });
-        } else if (_notEnoughFunds()) {
-          setState(() {
-            _errorAmountText = 'Not enough funds';
-          });
-        }
+  String? _validateAmount(String? input) {
+    String? errorText;
+    if (_notEnoughFunds()) {
+      errorText = 'Not enough Funds';
+    }
+    errorText = errorText ?? validateWitValue(input);
+    try {
+      if (num.parse(input!) == 0) {
+        errorText = errorText ?? 'Amount cannot be zero';
       }
+    } catch (e) {
+      errorText = 'Invalid Amount';
     }
-    return _errorAmountText != null ? false : true;
+    return errorText;
   }
 
-  bool _validateFee({force = false}) {
-    if (this.mounted && _isAbsoluteFee(_feeType)) {
-      if (force || _notFocusForm()) {
-        setState(() {
-          _errorFeeText = null;
-        });
-        if (_minerFee.isEmpty) {
-          setState(() {
-            _errorFeeText = 'This field is required';
-          });
-        } else if (!_isValidNumber(_amount)) {
-          setState(() {
-            _errorFeeText = 'Invalid number';
-          });
-        } else if (_notEnoughFunds()) {
-          setState(() {
-            _errorFeeText = 'Not enough funds';
-          });
-        }
-      }
+  String? _validateFee(String? input) {
+    String? errorText;
+    if (_notEnoughFunds()) {
+      errorText = 'Not enough Funds';
     }
-    return _errorFeeText != null ? false : true;
+    if (_isAbsoluteFee()) {
+      errorText = errorText ?? validateWitValue(input);
+    }
+    return errorText;
   }
 
-  bool validateForm({force = false}) {
-    if (_validateAddress(force: true) &&
-        _validateAmount(force: true) &&
-        _validateFee(force: true)) {
+  bool validateForm() {
+    setState(() {
+      _errorAddressText = _validateAddress(_address);
+      _errorAmountText = _validateAmount(_amount);
+      _hasAddressError = _errorAddressText != null;
+      _hasAmountError = _errorAmountText != null;
+      if (_feeType == FeeType.Absolute) {
+        _errorFeeText = _validateFee(_minerFee);
+        _hasFeeError = _errorFeeText != null;
+      }
+    });
+    if ((!_hasAddressError && !_hasAmountError && !_hasFeeError) &&
```

Comment:
You can avoid using _hasAddressError, _hasAmountError and _hasFeeError 
```suggestion
    if ((_errorAddressText == null && _errorAmountText == null && _errorFeeText == null) &&
```